EIPS/eip-6789.md

```diff
@@ -0,0 +1,88 @@
+---
+eip: 6789
+title: Rename `gas` to `mana`
+description: This EIP picks up on Vitalik Buterin's (@vbuterin) original proposal in 2015 to rename `gas` to `mana`
+author: Pascal Caversaccio (@pcaversaccio)
+discussions-to: https://ethereum-magicians.org/t/eip-6789-rename-gas-to-mana/13570
+status: Draft
+type: Standards Track
+category: Interface
+created: 2023-03-27
+---
+
+## Abstract
+
+This EIP picks up on Vitalik Buterin's (@vbuterin) original proposal in 2015 to rename `gas` to `mana`.
+
+## Motivation
+
+Vitalik's original motivation is threefold and is still valid today:
+
+- `mana` more accurately reflects the increased environmental friendliness of Proof-of-Stake (PoS),
+- `mana` is generally understood to be ephemeral and non-transferable, and thus represents accurately the characteristic of how `gas` is actually used within Ethereum, and
+- `mana` is generally considered renewable, whilst (natural) gas is considered non-renewable.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+The core term `gas` MUST be renamed to `mana`. This MUST also include renaming the opcode `Gas` to `Mana`.
```

Comment:
```suggestion
The core term `gas` MUST be renamed to `mana`.

The following opcodes MUST be renamed:

- `GASPRICE` to `MANAPRICE`;
- `GASLIMIT` to `MANALIMIT`;
- `GAS` to `MANA`

Additionally, the parameters/outputs of the following opcodes MUST be renamed:

- `CALL`'s `gas` parameter to `mana`
- `CALLCODE`'s `gas` parameter to `mana`
- `DELEGATECALL`'s `gas` parameter to `mana`
- `STATICCALL`'s `gas` parameter to `mana`
- `GASLIMIT`'s `gasLimit` output to `manaLimit`
- `GAS`'s `gas` output to `mana`

Finally, the following RPC endpoints MUST be renamed:

- `eth_estimateGas` to `eth_estimateMana`
- `eth_gasPrice` to `eth_manaPrice`
- `eth_maxPriorityFeePerGas` to `eth_maxPriorityFeePerMana`
```
I think this specification is more complete.

Reply:
Yes fully agreed. I was not sure in first place, whether to use an _implied_ specs or an explicit one. This one is much more clear! Thx.